### PR TITLE
Server Admin page for paid subscriptions

### PR DESCRIPTION
**Pull request by SailingSteve**
Create, edit, & delete paid subscriptions for organizations (coupons)
New PlanDialog.js, is heavily based on TaskDialog.js.

Work towards wevote/WebApp/issues/2488

### CODE REVIEW

File: apis_v1/views/views_donation.py [hunk 1]
```diff
@@ -229,3 +229,65 @@ def validate_coupon_for_api_view(request):
...
+    list_price_monthly_credit = request.GET.get('listPriceMonthlyCredit')
+    discounted_price_monthly_credit = request.GET.get('discountedPriceMonthlyCredit')
+    features_provided_bitmap = request.GET.get('featuresProvidedBitmap')
+    coupon_expires_date = request.GET.get('couponExpiresDate', None)
+    if len(coupon_expires_date) is 0:
+        coupon_expires_date = None
+    print("create_new_plan_for_api_view, plan_type_enum: " + plan_type_enum + ", coupon_code: " + coupon_code)
+    plan_on_stage = 0
+
+    voter_device_id = get_voter_device_id(request)  # We standardize how we take in the voter_device_id
```
Comment (DaleMcGrew) [inline apis_v1/views/views_donation.py:247]: Please also restrict this to 'admin' only, using this code (I will add a 'account_manager' classification later):
```
    # admin, partner_organization, political_data_manager, political_data_viewer, verified_volunteer
    authority_required = {'admin'}
    if not voter_has_authority(request, authority_required):
```

File: apis_v1/views/views_donation.py [hunk 2]
```diff
@@ -229,3 +229,65 @@ def validate_coupon_for_api_view(request):
...
+    json_data = {
+        'success': positive_value_exists(plan_on_stage.id),
+        'status': status,
+        'id': plan_on_stage.id if positive_value_exists(plan_on_stage.id) else 0.
+        }
+
+    return HttpResponse(json.dumps(json_data), content_type='application/json')
+
+def delete_plan_for_api_view(request):
+    id = request.GET.get('id')
```
Comment (DaleMcGrew) [inline apis_v1/views/views_donation.py:272]: Please also restrict this to 'admin' only, using this code:
```
    # admin, partner_organization, political_data_manager, political_data_viewer, verified_volunteer
    authority_required = {'admin'}
    if not voter_has_authority(request, authority_required):
```

File: apis_v1/static/js/PlanDialog.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        couponAppliedMessage: couponAppliedMessage.val(),
+        hiddenPlanComment: hiddenPlanComment.val(),
+        listPriceMonthlyCredit: dollarsToCents(listPriceMonthlyCredit.val()),
+        discountedPriceMonthlyCredit: dollarsToCents(discountedPriceMonthlyCredit.val()),
+        featuresProvidedBitmap: featuresProvidedBitmap.val(),
+        couponExpiresDate: expires,
+      }
+
+      const apiURL = `${window.location.origin}/apis/v1/createNewPlan`;
+      $.getJSON(apiURL, newPlanData, () => { dialog.dialog('close'); location.reload(); })
```
Comment (DaleMcGrew) [inline apis_v1/static/js/PlanDialog.js:158]: Does this (and `deleteThisPlan` below) API call include the voter_device_id cookie value so we can verify that the person doing the action has sufficient authority?

**Comment by DaleMcGrew** [comment]
Thank you @SailingSteve, I'm looking forward to playing with this! 👍

File: apis_v1/views/views_donation.py (hunk 1 above)
Comment (SailingSteve) [reply apis_v1/views/views_donation.py:247]: Done!

File: apis_v1/views/views_donation.py (hunk 2 above)
Comment (SailingSteve) [reply apis_v1/views/views_donation.py:272]: Done!

File: apis_v1/static/js/PlanDialog.js (hunk 3 above)
Comment (SailingSteve) [reply apis_v1/static/js/PlanDialog.js:158]: Yes, all Dispatcher.loadEndpoint() calls include the device_id cookie

**Pull request merged**